crates/paralegal-policy/src/context.rs

```diff
@@ -40,10 +40,118 @@ pub enum EdgeType {
     DataAndControl,
 }
 
+/// Data type representing nodes in the SPDG for a particular controller.
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Node<'a> {
+    /// ControllerId of the node.
+    pub ctrl_id: ControllerId,
+    /// The data of the node.
+    pub node: NodeType<'a>,
+}
+
+/// Enum identifying the different types of nodes in the SPDG
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub enum NodeType<'a> {
+    /// Corresponds to [`DataSource::Argument`].
+    CtrlArgument(&'a DataSource),
```

Comment:
Can we name this `ControllerArgument`? I just found myself very confused at a later point thinking this is control flow. 😅  (This isn't your fault, I know Will and I started this confusing naming convention where everything is `ctrl` but I think it's time we slowly disambiguate it)

Reply:
Haha sure